tools/deployment-cli-tools/ch_cli_tools/dockercompose.py

```diff
@@ -0,0 +1,891 @@
+        # {{ $values_copy | toYaml | indent 4 }}
+        for key, val in helm_values['apps'].items():
+            helm_values['apps'][key]['harness']['secrets'] = {}
+        return helm_values
+
+    def create_tls_certificate(self, helm_values):
+        if not self.tls:
+            helm_values['tls'] = None
+            return
+        if not self.local:
+            return
+        helm_values['tls'] = self.domain.replace(".", "-") + "-tls"
+
+        bootstrap_file = 'bootstrap.sh'
+        certs_parent_folder_path = self.output_path / 'helm' / 'resources'
+        certs_folder_path = certs_parent_folder_path / 'certs'
+
+        # if os.path.exists(os.path.join(certs_folder_path)):
+        if certs_folder_path.exists():
+            # don't overwrite the certificate if it exists
+            return
+
+        try:
+            client = DockerClient()
+            client.ping()
+        except:
+            raise ConnectionRefusedError(
+                '\n\nIs docker running? Run "eval(minikube docker-env)" if you are using minikube...')
+
+        # Create CA and sign cert for domain
+        container = client.containers.run(image='frapsoft/openssl',
+                                          command=f'sleep 60',
+                                          entrypoint="",
+                                          detach=True,
+                                          environment=[
+                                              f"DOMAIN={self.domain}"],
+                                          )
+
+        container.exec_run('mkdir -p /mnt/vol1')
+        container.exec_run('mkdir -p /mnt/certs')
+
+        # copy bootstrap file
+        cur_dir = os.getcwd()
+        os.chdir(Path(HERE) / 'scripts')
+        tar = tarfile.open(bootstrap_file + '.tar', mode='w')
+        try:
+            tar.add(bootstrap_file)
+        finally:
+            tar.close()
+        data = open(bootstrap_file + '.tar', 'rb').read()
+        container.put_archive('/mnt/vol1', data)
+        os.chdir(cur_dir)
+        container.exec_run(f'tar x {bootstrap_file}.tar', workdir='/mnt/vol1')
+
+        # exec bootstrap file
+        container.exec_run(f'/bin/ash /mnt/vol1/{bootstrap_file}')
+
+        # retrieve the certs from the container
+        bits, stat = container.get_archive('/mnt/certs')
+        if not certs_folder_path.exists():
+            certs_folder_path.mkdir(parents=True)
+        certs_tar = certs_parent_folder_path / 'certs.tar'
+        with open(certs_tar, 'wb') as f:
+            for chunk in bits:
+                f.write(chunk)
+        cf = tarfile.open(certs_tar)
+        cf.extractall(path=certs_parent_folder_path)
+
+        logs = container.logs()
+        logging.info(f'openssl container logs: {logs}')
+
+        # stop the container
+        container.kill()
+
+        logging.info("Created certificates for local deployment")
+
+    def __finish_helm_values(self, values):
+        """
+        Sets default overridden values
```

Comment:
this logic is the same as in helm.py, extracting a superclass might allow us to avoid duplication